src/bit_array.cr

```diff
@@ -1,22 +1,51 @@
+# BitArray is an array data structure that compactly stores bits.
+#
+# Bits externally represented as `Bool`s and are stored internally as
+# `UInt32`s. The total number of bits stored is set at creation and is
+# immutable.
+#
+# `BitArray` includes all the methods in `Enumerable`.
```

Comment:
dunno what kind of wording we want to use, in general I personally fancy "x behaves like a y"